src/component.cpp

```diff
@@ -139,10 +139,27 @@ void Component::removeMath()
     mPimpl->mMath.clear();
 }
 
-void Component::addVariable(const VariablePtr &variable)
+bool Component::addVariable(const VariablePtr &variable)
 {
+    // Prevent adding null pointer.
+    if (variable == nullptr) {
+        return false;
+    }
+
+    // Prevent adding multiple times to list.
+    if (hasVariable(variable)) {
+        return false;
+    }
+
+    // Prevent adding to multiple components.
+    if (variable->hasParent()) {
+        auto otherParent = std::dynamic_pointer_cast<Component>(variable->parent());
+        otherParent->removeVariable(variable);
+    }
```

Comment:
This means that we are not adding a variable but moving it.

Reply:
That's what addComponent does ... see Andre's comment earlier.

Reply:
https://github.com/cellml/libcellml/pull/621#discussion_r437822306